pkg/server/api/bundle/v1/service.go

```diff
@@ -161,17 +155,85 @@ func (s *Service) GetFederatedBundle(ctx context.Context, req *bundle.GetFederat
 		return nil, status.Errorf(codes.NotFound, "bundle for %q not found", req.TrustDomain)
 	}
 
-	b, err := applyMask(dsResp.Bundle, req.OutputMask)
+	b, err := api.BundleToProto(dsResp.Bundle)
 	if err != nil {
-		log.Errorf("Failed to apply mask: %v", err)
-		return nil, status.Errorf(codes.Internal, "failed to apply mask: %v", err)
+		log.WithError(err).Error("Failed to convert bundle")
+		return nil, status.Errorf(codes.Internal, "failed to convert bundle: %v", err)
 	}
 
+	applyBundleMask(b, req.OutputMask)
+
 	return b, nil
 }
 
 func (s *Service) BatchCreateFederatedBundle(ctx context.Context, req *bundle.BatchCreateFederatedBundleRequest) (*bundle.BatchCreateFederatedBundleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method BatchCreateFederatedBundle not implemented")
+	log := rpccontext.Logger(ctx)
+
+	results := []*bundle.BatchCreateFederatedBundleResponse_Result{}
```

Comment:
what happens when req is emtpy? must we do nothing? or return an error?

Reply:
Hmm, good question. I think I'd lean towards returning an empty response instead of failing.

Also, I'd probably recommend either initializing this slice to nil, or preallocating the slice to the number of incoming bundles.